framework/py/flwr/serverapp/strategy/dp_adaptive_clipping.py

```diff
@@ -0,0 +1,343 @@
+    ) -> Iterable[Message]:
+        """Configure the next round of federated evaluation."""
+        return self.strategy.configure_evaluate(server_round, arrays, config, grid)
+
+    def aggregate_evaluate(
+        self, server_round: int, replies: Iterable[Message]
+    ) -> Optional[MetricRecord]:
+        """Aggregate MetricRecords in the received Messages."""
+        return self.strategy.aggregate_evaluate(server_round, replies)
+
+    def summary(self) -> None:
+        """Log summary configuration of the strategy."""
+        self.strategy.summary()
+
+
+class DifferentialPrivacyServerSideAdaptiveClipping(DifferentialPrivacyAdaptiveBase):
+    """Message-based central DP with server-side adaptive clipping."""
+
+    # pylint: disable=too-many-arguments,too-many-locals,too-many-positional-arguments
+    def __init__(
+        self,
+        strategy: Strategy,
+        noise_multiplier: float,
+        num_sampled_clients: int,
+        initial_clipping_norm: float = 0.1,
+        target_clipped_quantile: float = 0.5,
+        clip_norm_lr: float = 0.2,
+        clipped_count_stddev: Optional[float] = None,
+    ) -> None:
+        super().__init__(
+            strategy,
+            noise_multiplier,
+            num_sampled_clients,
+            initial_clipping_norm,
+            target_clipped_quantile,
+            clip_norm_lr,
+            clipped_count_stddev,
+        )
+        self.current_arrays: ArrayRecord = ArrayRecord()
+
+    def __repr__(self) -> str:
+        """Compute a string representation of the strategy."""
+        return "Differential Privacy Strategy Wrapper (Server-Side Adaptive Clipping)"
+
+    def configure_train(
+        self, server_round: int, arrays: ArrayRecord, config: ConfigRecord, grid: Grid
+    ) -> Iterable[Message]:
+        """Configure the next round of training."""
+        self.current_arrays = arrays
+        return self.strategy.configure_train(server_round, arrays, config, grid)
+
+    def aggregate_train(
+        self, server_round: int, replies: Iterable[Message]
+    ) -> tuple[Optional[ArrayRecord], Optional[MetricRecord]]:
+        """Aggregate ArrayRecords and MetricRecords in the received Messages."""
+        if not self._validate_replies(replies):
+            return None, None
+
+        current_nd = self.current_arrays.to_numpy_ndarrays()
+        clipped_indicator_count = 0
+        replies_list = list(replies)
+
+        for reply in replies_list:
+            for arr_name, record in reply.content.array_records.items():
+                reply_nd = record.to_numpy_ndarrays()
+                model_update = [
+                    np.subtract(x, y) for (x, y) in zip(reply_nd, current_nd)
+                ]
+                norm_bit = adaptive_clip_inputs_inplace(
+                    model_update, self.clipping_norm
+                )
+                clipped_indicator_count += int(bool(norm_bit))
```

Comment:
```suggestion
                clipped_indicator_count += int(norm_bit)
```
I think we don't need it as `norm_bit` is already a boolean?